extensions/csv/src/main/java/io/deephaven/csv/parsers/DoubleParser.java

```diff
@@ -0,0 +1,58 @@
+package io.deephaven.csv.parsers;
+
+import io.deephaven.csv.sinks.Sink;
+import io.deephaven.csv.tokenization.Tokenizer;
+import io.deephaven.csv.parsers.context.ParseContext;
+
+public final class DoubleParser extends ParserBase<double[]> {
+    public static final DoubleParser INSTANCE = new DoubleParser();
+
+    private DoubleParser() {}
+
+    @Override
+    public Sink<double[]> tryParse(ParseContext ctx, IteratorHolder ih, IteratorHolder ihAlt) {
+        return twoPhaseDriver(ctx, ih, ihAlt, DoubleParser::tryParseHelper, ctx.sinkFactory::makeDoubleSink);
+    }
+
+    private static boolean tryParseHelper(ParseContext ctx, IteratorHolder ih, Sink<double[]> sink,
+                                          long current, long end) {
+        final double[] chunk = new double[DEST_BLOCK_SIZE];
+        final double[] doubleHolder = new double[1];
```

Comment:
Similarly MutableDouble as previous comment, and you can extrapolate to any other uses.